src/views/action/stores/actionStore.js

```diff
@@ -0,0 +1,189 @@
+    }
+  },
+  /* action actions */
+  addAction ({ commit }) {
+    commit('resetCurrentAction')
+    commit('showModal', { modal: 'modal_action', status: true })
+  },
+  editAction ({ commit }, item) {
+    commit('setCurrentAction', item)
+    commit('showModal', { modal: 'modal_action', status: true })
+  },
+  async actionGetApi ({ state, commit }, request) {
+    commit('actionStore/paginationStore/setLoading', true, { root: true })
+    const response = await actionGetApi(request)
+    if ('payload' in response.data) {
+      commit('actionStore/paginationStore/init', response.data.payload, { root: true })
+    }
+  },
+  async deleteAction ({ state, commit, dispatch }, item) {
+    const res = await this._vm.$confirm('¿ Esta seguro de realizar esta acción ?', {
+      title: 'Confirmar',
+      color: 'error',
+      buttonFalseText: 'Cancelar',
+      buttonTrueText: 'Aceptar',
+      buttonTrueColor: 'error',
+      buttonFalseColor: 'info',
+      icon: 'mdi-alert-octagon',
+    })
+    if (!res) return
+    const response = await actionDeleteApi(item.id)
+    if ('payload' in response.data) {
+      response.data.payload && dispatch('actionGetApi', state.paginationStore.pagination_request)
+      commit('globalStore/errorSnackbar', response, { root: true })
+    } else {
+      commit('globalStore/errorSnackbar', response, { root: true })
+    }
+  },
+  async saveAction ({ state, commit, dispatch }) {
+    commit('setProgressLoading', { name: 'loading_modal_action', status: true })
+    const response = state.current_action_index === null ? await actionCreateApi(state.current_action) : await actionPatchApi(state.current_action)
+    if ('payload' in response.data) {
+      commit('setProgressLoading', { name: 'loading_modal_action', status: false })
+      commit('showModal', { modal: 'modal_action', status: false })
+      commit('resetCurrentAction')
+      dispatch('actionGetApi', state.paginationStore.pagination_request)
+      commit('globalStore/errorSnackbar', response, { root: true })
+    } else {
+      commit('setProgressLoading', { name: 'loading_modal_action', status: false })
+      commit('globalStore/errorSnackbar', response, { root: true })
+    }
+  },
+  formPredicates ({ state, commit }, res) {
+    // esto creo se debe optimizar y refactorizar de momento como es un solo campo del filtro lo deje asi.
+    const predicate = {
+      name: 'action_name',
+      comparison: 'contains',
+      attribute: 'action.name',
+      value: res,
+    }
+    const currentPredicates = state.paginationStore.datatable_options.predicates
+    if (currentPredicates.length) {
+      const key = currentPredicates.map(e => { return e.name }).indexOf(predicate.name)
+      commit('setPredicate', { predicate, type: 'update', key })
+    } else {
+      commit('setPredicate', { predicate, type: 'add' })
+    }
+    // hice esta mutation por que modificar predicates dentro de  datatable_options con  la mutation setPredicate
```

Comment:
creo que para que esto funcione tienes que user Vue.set
https://www.telerik.com/blogs/so-what-actually-is-vue-set

Reply:
utilice el $set cuando es actualizacion del item del predicates, es lo unico que no funcionaba. 
las opciones push, splice de un array si desencadena la reactividad.